### ISSUE TITLE
Bare text preceded by a mountable sibling is a real insertBefore hazard that neither rule can see

### ISSUE THREAD

**Issue by Faithfinder** [getcouped/eslint-plugin-react-google-translate#7]
Both rules fire only on `ConditionalExpression` and `LogicalExpression`. That misses a genuine `insertBefore` hazard, which means a green lint cannot be read as "these rules found nothing dangerous".

Filed separately from #6 (which fixes four false-positive classes) because fixing this changes reporting volume rather than correcting a wrong answer, so it needs a judgement call from you rather than a patch smuggled into a precision PR.

## Reproduction

```jsx
function NavFilterTile({ selectedItem, isLoading }) {
  return (
    <div>
      {selectedItem?.icon}
      {selectedItem?.label}      {/* bare HostText, no conditional anywhere */}
      {isLoading && <ProgressBar />}
    </div>
  );
}
```

Lints clean today, with `projectService` enabled and both rules set to `error`.

## Why it throws

When `selectedItem?.icon` goes from absent to present, React mounts it and calls `getHostSibling` to find the reference node for `insertBefore`. `getHostSibling` searches **forward**, so it lands on the text node created for `selectedItem?.label`. Google Translate has reparented that text node under a `<font>` wrapper, while React's `HostText` fiber still holds the original node — so `insertBefore(parent, icon, staleTextNode)` throws `NotFoundError: The child can not be found in the parent`.

Confirmed on react-dom 18.3.1 by mounting with `icon` absent, wrapping every text node in a `<font>`, then re-rendering with `icon` present:

```
{icon?}{bareLabel}{loading && <progress/>}   1 HostText fiber   NotFoundError: The child can not be found in the parent.
```

Note this is the `insertBefore` message, distinct from the `removeChild` message (`The node to be removed is not a child of this node`) that the conditional cases produce. The hazard is on **mount** of the preceding sibling; the same shape does not throw on unmount, consistent with `getHostSibling` being forward-only.

Neither `{selectedItem?.icon}` nor `{selectedItem?.label}` is a conditional expression, so no visitor examines either one. The rule's existing `conditionalSiblingsPrecedeNode` has the right idea and the right direction — it is just gated on the preceding sibling being a `ConditionalExpression` or `LogicalExpression`, and on the node itself arriving through a conditional.

## Suggested shape of a fix

Two independent widenings, both reusing machinery #6 already introduces:

1. **What counts as text at a position.** Apply the `text-node-preceded-by-conditional` check to any child that contributes text — including a `MemberExpression` or `ChainExpression` whose type resolves to `string`/`number` — not only static `JSXText`. #6 already moves this check onto expression containers; this extends which contributions qualify.

2. **What counts as a preceding sibling that can mount.** Widen beyond conditionals to include optional-chained member expressions, which are syntactically visible as possibly-absent. `{selectedItem?.icon}` can render nothing when `selectedItem` is nullish, which is exactly the mount that triggers the hazard.

Both need type information to stay precise, so this is only viable in the type-aware path. Reporting every `UNKNOWN` contribution instead would flag every `ReactNode`-typed value, which is the outcome you ruled out in #1 — I don't think that trade is worth making.

I'd expect this to increase report volume on real codebases, since the shape is common. Two options if that's a concern: a separate rule (`no-unwrapped-text-nodes-with-mountable-siblings` or similar) so it can be adopted independently, or an option on the existing rule. Happy to implement either once #6 is settled — I didn't want to guess which you'd prefer.

## Why it matters for the rules' contract

The two things interact. Improving precision (#6) without also improving recall makes the rules *more* trustworthy without making them more complete, and users reasonably read a clean run as "no Translate hazards here". Right now that inference isn't sound, and the README doesn't say so — #6 adds a note under Known limitations, but a documented gap is a workaround, not a fix.
